tools/walletextension/wallet_extension.go

```diff
@@ -348,6 +400,86 @@ func setCallFromFieldIfMissing(args []interface{}, account common.Address) ([]in
 	return args, nil
 }
 
+// Stores a viewing key to disk.
+func (we *WalletExtension) persistViewingKey(viewingKey *rpcclientlib.ViewingKey) {
+	viewingPrivateKeyBytes := crypto.FromECDSA(viewingKey.PrivateKey.ExportECDSA())
+
+	persistenceEntry := fmt.Sprintf("%s %s %s %s",
+		we.hostAddr,
+		viewingKey.Account.Hex(),
+		// We encode the bytes as hex to ensure there are no unintentional line breaks to make parsing the file harder.
+		hex.EncodeToString(viewingPrivateKeyBytes),
+		hex.EncodeToString(viewingKey.SignedKey))
+
+	persistenceFile, err := os.OpenFile(we.persistencePath, os.O_APPEND|os.O_WRONLY, 0o644)
+	defer persistenceFile.Close() //nolint:staticcheck
+	if err != nil {
+		log.Error("could not open persistence file. Cause: %s", err)
+	}
+
+	_, err = fmt.Fprintln(persistenceFile, persistenceEntry)
+	if err != nil {
+		log.Error("failed to write viewing key to persistence file. Cause: %s", err)
+	}
+}
+
+// Loads any viewing keys from disk. Viewing keys for other hosts are ignored.
+func (we *WalletExtension) loadViewingKeys() map[common.Address]*rpcclientlib.ViewingKey {
+	viewingKeys := make(map[common.Address]*rpcclientlib.ViewingKey)
+
+	persistenceFile, err := os.OpenFile(we.persistencePath, os.O_RDONLY, 0o644)
+	defer persistenceFile.Close() //nolint:staticcheck
+	if err != nil {
+		log.Error("could not open persistence file. Cause: %s", err)
+	}
+
+	scanner := bufio.NewScanner(persistenceFile)
+	for scanner.Scan() {
+		// TODO - Determine strategy for invalid persistence entries - delete? Warn? Shutdown? For now, we log a warning.
+		entry := scanner.Text()
```

Comment:
Did you consider using encoding/csv or even encoding/json to serialize this btw? Might make this marginally more robust/easier